<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [2]:
# Membuat dataframes dari dictionary
import pandas as pd

df = pd.DataFrame({'Nama':['Upin','Ipin','Mail','Fizi','Ehsan','Mei mei','Jarjit'],'Nilai Ujian':[75,70,85,65,90,95,80]})
print(df)

      Nama  Nilai Ujian
0     Upin           75
1     Ipin           70
2     Mail           85
3     Fizi           65
4    Ehsan           90
5  Mei mei           95
6   Jarjit           80


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [4]:
# Membaca data dan memasukannya kedalam suatu variabel
import pandas as pd
data_diamond = pd.read_csv('diamonds-1.csv')
print(data_diamond)

       Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
5               6   0.24  Very Good     J    VVS2   62.8   57.0    336  3.94   
6               7   0.24  Very Good     I    VVS1   62.3   57.0    336  3.95   
7               8   0.26  Very Good     H     SI1   61.9   55.0    337  4.07   
8               9   0.22       Fair     E     VS2   65.1   61.0    337  3.87   
9              10   0.23  Very Good     H     VS1   59.4   61.0    338  4.00   
10             11   0.30       Good     J     SI1   64.0   55.0    339  4.25   
11             12   0.23      Ideal     

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

<h4> Jawab: </h4>
- Tipe data pada Series harus seragam, sedangkan tipe data pada list & dictionary dapat berbeda-beda
- Dataframe berisi data dua dimensi (kolom dan baris), sedangkan Series hanya berisi data 1 dimensi


In [5]:
# code here
# Contoh data series
series = pd.Series(['Taeil', 'Johnny', 'Taeyong','Yuta','Doyoung', 'Jaehyun', 'Jungwoo', 'Mark', 'Haechan'])
print('Daftar member NCT127 : ')
print(series.to_string(index=False))

Daftar member NCT127 : 
   Taeil
  Johnny
 Taeyong
    Yuta
 Doyoung
 Jaehyun
 Jungwoo
    Mark
 Haechan


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di quiz no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [6]:
# code here
first = data_diamond.head(1)
last = data_diamond.tail(1)
print('Data baris pertama\n',first,'\n')
print('Data baris terakhir\n',last,'\n')

print('Data baris pertama dari kolom price adalah %d'% first['price'])
print('Data baris terakhir dari kolom color adalah %s'% last['color'].to_string(index=False)) 
# Nilai data baris pertama dari kolom price adalah 326
# Nilai data baris terakhir dari kolom color adalah D


Data baris pertama
    Unnamed: 0  carat    cut color clarity  depth  table  price     x     y  \
0           1   0.23  Ideal     E     SI2   61.5   55.0    326  3.95  3.98   

      z  
0  2.43   

Data baris terakhir
        Unnamed: 0  carat    cut color clarity  depth  table  price     x  \
53939       53940   0.75  Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
53939  5.87  3.64   

Data baris pertama dari kolom price adalah 326
Data baris terakhir dari kolom color adalah  D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

<h4> Jawab: </h4>
Deskriptif statistik adalah metode yang berhubungan dengan pengumpulan atau penyajian data sampai memberi informasi yang berguna, contohnya:

- tabel
- diagram
- grafik
- besaran-besaran lain dalam majalah dan koran-koran


informasi yang didapatkan antara lain ukuran pemusatan data (mean, median, modus), ukuran penyebaran data, dan kecenderungan gugus data

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [7]:
# code here
desc_stats = data_diamond.describe()
print('Deskriptif statistik data diamond: \n',desc_stats,'\n')
price_mn = data_diamond['price'].mean()
print('Nilai rata-rata dari column price: ',round(price_mn,4))
depth_std = data_diamond['depth'].std()
print('Nilai standar deviasi dari column depth: ',round(depth_std,4))
carat_max = data_diamond['carat'].max()
print('Nilai maximum dari column carat: ',round(carat_max,4))

Deskriptif statistik data diamond: 
          Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
2

- Nilai rata-rata dari column price:  3932.7997
- Nilai standar deviasi dari column depth:  1.4326
- Nilai maximum dari column carat:  5.01